lib/internal/types.js

```diff
@@ -0,0 +1,32 @@
+'use strict';
+
+const ReflectApply = Reflect.apply;
+
+function uncurryThis(func) {
+  return (thisArg, ...args) => ReflectApply(func, thisArg, args);
```

Comment:
The args will always be a empty array. So you might just use a empty array that you pass in for the `argumentsList` instead of creating a new array each time?

Reply:
The spread syntax is actually faster than hardcoding an empty array by a good amount. Probably because of the fact that Reflect.apply is inlined.

Reply:
Interesting and good to know. In that case a comment about that would be good as well.

Reply:
@BridgeAR comment added.